IndustrialUnitProvider/UnitMapper.cs

```diff
@@ -0,0 +1,99 @@
+﻿using IndustrialUnit.Model.Model;
+using IndustrialUnitDatabase;
+using OfficeOpenXml;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace IndustrialUnitProvider
+{
+  public class UnitMapper
+  {
+    public void LoadUnitsFromSheet(string file, ref List<string> logMessage)
+    {
+      List<Equipment> equipments = new List<Equipment>();
+      var sheetEquipment = ExcelWorker.ReadExcel(file, ValidSheetNames.Equipment.ToString(), ref logMessage);
+      if(sheetEquipment != null)
+      AssignValue(equipments, sheetEquipment, ref logMessage);
+
+      List<Valve> valves = new List<Valve>();
+      var sheetValve = ExcelWorker.ReadExcel(file, ValidSheetNames.Valve.ToString(), ref logMessage);
+      if(sheetValve != null)
+      AssignValue(valves, sheetValve, ref logMessage);
+
+      List<Instrument> instruments = new List<Instrument>();
+      var sheetInstruments = ExcelWorker.ReadExcel(file, ValidSheetNames.Instrument.ToString(), ref logMessage);
+      if(sheetInstruments != null)
+      AssignValue(instruments, sheetInstruments, ref logMessage);
+    }
+
+    public void AssignValue<T>(List<T> parameterCollection, ExcelWorksheet sheet, ref List<string> logMessage) where T : class, new()
+    {
+      PropertyInfo[] properties = typeof(T).GetProperties();
+
+      var validation = new ExcelValidation();
+      var columnNameToIndex = validation.CollectColumnNamesFromExcel(sheet);
+
+      if (!validation.ValidateColumnNames(columnNameToIndex, properties, sheet, ref logMessage))  return;
+
+      for (int rowIndex = 2; rowIndex < sheet.Dimension.Rows + 1; rowIndex++)
+      {
+        if (!string.IsNullOrEmpty(sheet.Cells[rowIndex, 2].Text)) //if an ItemType cell is empty in the source excel file item won't be added
+        {
+          T unit = new T();
+          foreach (var item in properties)
+          {
+            if (columnNameToIndex.TryGetValue(item.Name, out int value) && item.Name != "Id")
```

Comment:
Azért is lehet lassú, mert minden sorban minden property-hez újra megkeresed, hogy melyik oszlopban van. Ezt egyszer elég lenne megcsinálni, és összerendelni egy listában a (Property, Index) párokat. Utána foreach erre a listára, és `cellText = sheet.Cells[rowIndex, item.Index].Text`.